File: laravel-api/app/Models/TimeZone.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class TimeZone extends Model
{
    protected $fillable = [
        'name',
        'start_time',
        'end_time',
    ];

    protected $casts = [
        'start_time' => 'datetime',
        'end_time' => 'datetime',
    ];

    /**
     * Get the time slot instances for this time zone.
     */
    public function timeSlotInstances(): HasMany
    {
        return $this->hasMany(TimeSlotInstance::class);
    }
} 

File: flutter_app/lib/models/TimeZone.dart
class TimeZone {
  final int id;
  final String name;
  final String startTime;
  final String endTime;

  TimeZone({
    required this.id,
    required this.name,
    required this.startTime,
    required this.endTime,
  });

  factory TimeZone.fromJson(Map<String, dynamic> json) {
    return TimeZone(
      id: json['id'],
      name: json['name'],
      startTime: json['start_time'],
      endTime: json['end_time'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'start_time': startTime,
      'end_time': endTime,
    };
  } 
}
